### ISSUE TITLE
Iframe's inner body content is removed even with phx-update="ignore"

### ISSUE THREAD

**Issue by ajacquierbret** [phoenixframework/phoenix_live_view#2889]
### Environment

* Elixir version (elixir -v):
`Erlang/OTP 24 [erts-12.3] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] / Elixir 1.14.4 (compiled with Erlang/OTP 23)`

* Phoenix version (mix deps):
`phoenix 1.7.9`

* Phoenix LiveView version (mix deps):
`phoenix_live_view 0.18.18`

* Operating system:
`macOS 13.0.1`

* Browsers you attempted to reproduce this bug on (the more the merrier): 
`Safari 16.1` | `Google Chrome 118`

* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no:
YES

### Actual behavior

I'm trying to render a TinyMCE iframe through a JS hook with Phoenix LiveView. Because dom patching removes the iframe node from the dom on every render, I have to set `phx-update="ignore"` on the parent node of the hook. It works well until another unrelated node is added to the dom via an `:if={@show}` attribute, then the content of the iframe's inner body tag is removed, even though my guess is that it should not.

```ex
# EditorHook.js

const EditorHook = {
    content() {
        return this.el.dataset.content;
    },
    mounted() {
        const scope = this;

        tinymce.init({
                selector: `#${this.el.id}`,
                setup: editor => {
                    editor.on('init', () => {
                        editor.setContent(scope.content());
                    })
                }
            )
        }
    }
}
```

```ex
# index.html.heex

<.modal
  :if={@show}
  id="exit-confirmation-modal"
  show={@show}
  on_cancel={JS.push("show_exit_confirmation_modal", value: %{show: false})}
>
...
</.modal>
...
<div id={"editor-#{@id}-container"} phx-update="ignore">
    <div
      id={"editor-#{@id}"}
      phx-hook="EditorHook"
      data-content="<span>Hello, World.</span>"
    />
</div>
```

Here's the rendered HTML before `assign(socket, show: true)`:
```html
<div id="editor-2-container" class="pt-5" phx-update="ignore">
   <div id="editor-2" phx-hook="EditorHook" data-content="<span>Hello, World.</span>" style="display: none;" aria-hidden="true"></div>
   <div role="application" class="tox tox-tinymce" aria-disabled="false" style="visibility: hidden; height: 650px;">
      <div class="tox-editor-container">
         <div class="tox-edit-area">
            <iframe id="editor-2_ifr" frameborder="0" allowtransparency="true" title="Zone de Texte Riche" class="tox-edit-area__iframe" srcdoc="<!DOCTYPE html><html><head><meta http-equiv=&quot;Content-Type&quot; content=&quot;text/html; charset=UTF-8&quot; /></head><body id=&quot;tinymce&quot; class=&quot;mce-content-body &quot; data-id=&quot;editor-2&quot; aria-label=&quot;Zone de Texte Riche. Appuyez sur ALT-0 pour l'aide.&quot;><br></body></html>">
              #document
              <!DOCTYPE html>
               <html>
                  <head>
                     <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
                     <link rel="stylesheet" type="text/css" id="mce-u0" href="http://localhost:4000/assets/skins/ui/oxide/content.min.css">
                     <link rel="stylesheet" type="text/css" id="mce-u1" href="http://localhost:4000/assets/skins/content/default/content.css">
                  </head>
                  <body id="tinymce" class="mce-content-body " data-id="editor-2" aria-label="Zone de Texte Riche. Appuyez sur ALT-0 pour l'aide." contenteditable="true" spellcheck="false">
                     <p>Hello, World.</p>
                  </body>
               </html>
            </iframe>
         </div>
      </div>
   </div>
</div>
```

And after:
```html
<div id="editor-2-container" class="pt-5" phx-update="ignore">
   <div id="editor-2" phx-hook="EditorHook" data-content="<span>Hello, World.</span>" style="display: none;" aria-hidden="true"></div>
   <div role="application" class="tox tox-tinymce" aria-disabled="false" style="visibility: hidden; height: 650px;">
      <div class="tox-editor-container">
         <div class="tox-edit-area">
            <iframe id="editor-2_ifr" frameborder="0" allowtransparency="true" title="Zone de Texte Riche" class="tox-edit-area__iframe" srcdoc="<!DOCTYPE html><html><head><meta http-equiv=&quot;Content-Type&quot; content=&quot;text/html; charset=UTF-8&quot; /></head><body id=&quot;tinymce&quot; class=&quot;mce-content-body &quot; data-id=&quot;editor-2&quot; aria-label=&quot;Zone de Texte Riche. Appuyez sur ALT-0 pour l'aide.&quot;><br></body></html>">
              #document
              <!DOCTYPE html>
               <html>
                  <head>
                     <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
                  </head>
                  <body id="tinymce" class="mce-content-body " data-id="editor-2" aria-label="Zone de Texte Riche. Appuyez sur ALT-0 pour l'aide." contenteditable="true" spellcheck="false">
                     <br>
                  </body>
               </html>
            </iframe>
         </div>
      </div>
   </div>
</div>
```

The content of the iframe after the update is the value of the `srcdoc` attribute. When setting the editor to `inline_mode: true` which renders it without an iframe, dom patching doesn't have any effect.

### Expected behavior

I expect that `phx-update="ignore"` prevents the iframe content to be removed.

**Comment by josevalim** [comment]
Can you please provide a small app that reproduces the error? Here is a starting point: https://github.com/wojtekmach/mix_install_examples/blob/main/phoenix_live_view.exs

**Comment by ajacquierbret** [comment]
@josevalim Sure, here's a repro made with `mix phx.new`: https://github.com/ajacquierbret/issue_2889

Install deps with `cd assets && yarn`, run the app with `mix phx.server`, type or delete any character in the text editor and look at the iframe's `<body>` content. The app runs in a LiveView located at `lib/issue_2889_web/live/index.ex`.
I placed a commented `SocketOptions.dom` option as a workaround in `assets/js/app.js`, but it has drawbacks and I'd like to be sure that the behaviour I describe in this issue is normal before using the "workaround" in my app. 

Just for you to be sure about what's going on with the `EditorHook`:
It first receives its initial content through a `data-content` prop, initializes itself with it, and pushes a `"show"` event to the server on content update. The server then updates the `:show` assign to `true` which adds a `<div>` to the DOM and resets the iframe's content.

Thanks a lot for your time!

**Comment by SteffenDE** [comment]
So, I tried to look into this a little bit and I think the problem lies in the way browsers handle iframes when they are moved in the DOM. When the `div` is toggled, morphdom seems to create a new parent container and appends the new div as well as the old container with `phx-update="ignore"`. So while the contents of the ignored container do not change, the fact that it moved in the DOM causes the browser to reload the iframe. You can also see this behavior with iframes that have a `src`. When adding the div on top, the iframe is reloaded, even though it is inside a container that has `phx-update="ignore"`. I'm not sure why morphdom does not "just" append the div to the existing node. I added some logging to the LiveView code to illustrate this:

<img width="880" alt="image" src="https://github.com/phoenixframework/phoenix_live_view/assets/4116351/e6c5fd5b-87cf-4d79-8c39-dfbf74df6564">

I tried digging into morphdom to see what's going on, but I did not get any more insights from that. The bottom line is that the ignored container is added to a new parent in this line https://github.com/patrick-steele-idem/morphdom/blob/30ad218dece89c83611df414aa0bc41a8b12fb34/src/morphdom.js#L373 and this causes all iframes that are contained to be reloaded.

To sum up: this seems to be caused by the operations morphdom performs on the dom. I'm not sure if LiveView can do anything to prevent this :(

Someone with more insights into morphdom can probably clarify if this could be "fixed" or is expected.

**Comment by chrismccord** [comment]
Iframes can’t be repositioned in the DOM without being destroyed and recreated. This is unfortunately nothing we can special case on our side. Thanks!

**Comment by Rodeoclash** [comment]
This is fixed in Chrome 133 via https://chromestatus.com/feature/5135990159835136 although it doesn't look like Morphdom supports it.

**Comment by SteffenDE** [comment]
@Rodeoclash I think this is worth opening up an issue in the morphdom repo. Once this gets more widespread, morphdom should probably try to use it.

**Comment by Rodeoclash** [comment]
@SteffenDE Good idea! I've raised it here https://github.com/patrick-steele-idem/morphdom/issues/266
